File: algorithms/sorting/Python3/bubble_sort.py

import random


def bubble_sort_v1(uil):
    swapped = False
    while swapped:
        swapped = False
        for i in range(0, len(uil)-1):
            if uil[i+1] < uil[i]:
                uil[i], uil[i+1] = uil[i+1], uil[i]
                swapped = True
    return uil


if __name__ == "__main__":
    unsorted_integer_list = [random.randint(0, 101) for i in range(20)]
    sorted_integer_list = bubble_sort_v1(unsorted_integer_list[:])
    print("Unsorted: ", unsorted_integer_list)
    print("Sorted: ", sorted_integer_list);

File: algorithms/sorting/Java/BubbleSort.java
import java.util.Arrays;
import java.util.Random;

/*
 * Time Complexity:
 *   Omega(n)
 *   Theta(n^2)
 *   O(n^2)
 *
 * Space Complexity:
 *   O(n^2)
 */
class BubbleSort {
    public static void main(String[] args) {
        Random rand = new Random();
        int[] nums = new int[20];

        for (int i = 0; i < nums.length; ++i) {
            nums[i] = rand.nextInt(101);
        }
        sort1(nums);
    }

    public static void sort1(int[] arr) {
        System.out.println("Unsorted: " + Arrays.toString(arr));
        boolean swapped = true;
        while (swapped) {
            swapped = false;
            for (int i = 1; i < arr.length; ++i) {
                if (arr[i] < arr[i-1]) {
                    swap(arr, i-1, i);
                    swapped = true;
                }
            }
        }
        System.out.println("Sorted: " + Arrays.toString(arr));
    }

    public static void swap(int[] arr, int left, int right) {
        int temp = arr[left];
        arr[left] = arr[right];
        arr[right] = temp;
    }
}
